drivers/spi/spi_ll_stm32.c

```diff
@@ -618,6 +618,49 @@ static int spi_stm32_configure(const struct device *dev,
 		LL_SPI_SetDataWidth(spi, LL_SPI_DATAWIDTH_16BIT);
 	}
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	const uint32_t mssi_cycles[] = {
```

Comment:
These table are an expensive way to configure this. Also, it seems that values encoding looks fairly simple.
What about using directly dt values w/o these tables ?

Reply:
i totally aggree with you but following values shifting: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/58497719/72e76736-aeac-49cc-a30d-4f869198b77d)
so i didn't bit shift at the start but update for your suggestion